packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -1200,14 +1204,18 @@ export class ContainerRuntime extends TypedEventEmitter<IContainerRuntimeEvents>
         const electedSummarizerContent = JSON.stringify(this.summarizerClientElection.serialize());
         addBlobToSummary(summaryTree, electedSummarizerBlobName, electedSummarizerContent);
 
-        const snapshot = this.blobManager.snapshot();
+        const { ids, table } = this.blobManager.snapshot();
 
         // Some storage (like git) doesn't allow empty tree, so we can omit it.
         // and the blob manager can handle the tree not existing when loading
-        if (snapshot.entries.length !== 0) {
-            const blobsTree = convertToSummaryTree(snapshot, false);
+        if (ids.entries.length !== 0) {
+            const blobsTree = convertToSummaryTree(ids, false);
             addTreeToSummary(summaryTree, blobsTreeName, blobsTree);
         }
+
+        if (table) {
+            addBlobToSummary(summaryTree, blobRedirectTableBlobName, table);
+        }
```

Comment:
Can all of this logic (RE what is in the blob manager summary) be in blob manager layer, and this layer not caring about details?

Reply:
I think the only way to do this would be to move BlobManager loading out of the ContainerRuntime constructor into ContainerRuntime.load(), since the attachment types don't need to be read from storage but other blobs do.

Reply:
I think we should definitely consider that in general. I.e. move some of the logic in ContainerRuntime.load() and ContainerRuntime.constructor into async ContainerRuntime.initialize() and call it from load() as part of the loading flow.

Reply:
I've done this, please take a look